### DISCUSSION TITLE
Why do not you calculate the real output token for throughput?

### DISCUSSION THREAD

**Discussion by wqh17101** [vllm-project/vllm#493] [Q&A]
https://github.com/vllm-project/vllm/blob/c894836108732d0cbb6fce15aeda8de1218a380d/benchmarks/benchmark_throughput.py#L177-L180

You can see that you just use the tokens' length of request in the dataset , not the real output length for the model to eval.

**Comment by Godricly** [comment]
The token returned also contains input prompt. I don't think computing the output will solve you problem.
https://github.com/vllm-project/vllm/blob/c6dfc3cdbeb4687a8313b82d399177e293fee196/vllm/entrypoints/api_server.py#L64-L67

**Comment by zhuohan123** [comment]
I believe the main reason is that prompt tokens also take compute. There are different ways to measure throughput for LLMs, but I believe the trends between systems should be similar under different metrics. Move this issue to discussions for future questions.

**Reply by zhaoyang-star** [reply]
Yes. It is an open discussion whether or not take prompt tokens into account. As far as I know, the latency metric is not included the length of prompt tokens.
